.azure/pipelines/jobs/default-build.yml

```diff
@@ -79,13 +79,19 @@ jobs:
     ${{ if and(eq(parameters.poolName, ''), eq(parameters.agentOs, 'Linux')) }}:
       vmImage: ubuntu-18.04
     ${{ if and(eq(parameters.poolName, ''), eq(parameters.agentOs, 'Windows')) }}:
-      vmImage: windows-latest
+      vmImage: windows-2019
+      ${{ if ne(variables['System.TeamProject'], 'public') }}:
```

Comment:
I'm not sure this extra defining of pool name works, though I've definitely seen some astoundingly obfuscated yaml?  It seems like in the internal case you're adding it defines vmImage, name, and Demands, which might trip it up.

Reply:
It's working fine in our release/2.1 branch. I believe we've done similar things in other repositories too. It's weird, but I'm not going to mess with success at the moment. Might do a follow-up at some point.

The main oddity is in logging. From [internal build #20230213.2](https://dev.azure.com/dnceng/internal/_build/results?buildId=2112206&view=results):

``` text
Pool: [NetCore1ESPool-Svc-Internal](https://dev.azure.com/dnceng/7ea9116e-9fac-403d-b258-b31fcf1bb293/_settings/agentqueues?poolId=&queueId=392)
Image: windows-2019
Agent: NetCore1ESPool-Svc-Internal 34
...
Starting: Initialize job
Agent name: 'NetCore1ESPool-Svc-Internal 34'
Agent machine name: '1c059be9c000000'
Current agent version: '2.217.2'
1ES Hosted Pool
SKU: Standard_D4_v3
Image: windows.vs2019.amd64
Image Version: 2023.0126.180713
Interactive Mode: True
```

Reply:
#1024